src/lithium/interestingness/timed_run.py

```
@@ -46,72 +46,86 @@ def __init__(self,  # pylint: disable=missing-param-doc,missing-type-doc,too-man
         env["ASAN_SYMBOLIZER_PATH"] = os.path.join(symbolizer_path, "llvm-symbolizer")
     return env
 
 
-def timed_run(commandWithArgs, timeout, logPrefix, inp=None, preexec_fn=None):  # pylint: disable=invalid-name
-    # pylint: disable=missing-param-doc,missing-raises-doc,missing-return-doc,missing-return-type-doc,missing-type-doc
-    # pylint: disable=too-complex,too-many-branches,too-many-locals,too-many-statements
-    """If logPrefix is None, uses pipes instead of files for all output."""
-    if not isinstance(commandWithArgs, list):
-        raise TypeError("commandWithArgs should be a list (of strings).")
+def timed_run(cmd_with_args, timeout, log_prefix, env=None, inp=None, preexec_fn=None):
+    # pylint: disable=too-complex,too-many-arguments,too-many-branches,too-many-locals,too-many-statements
+    """If log_prefix is None, uses pipes instead of files for all output.
+
+    Args:
+        cmd_with_args (list): List of command and parameters to be executed
+        timeout (int): Timeout for the command to be run, in seconds
+        log_prefix (str): Prefix string of the log files
+        env (dict): Environment for the commmand to be executed in
+        inp (str): stdin to be passed to the command
+        preexec_fn (function): preexec_fn to be passed to subprocess.Popen
+
+    Raises:
+        TypeError: Raises if input parameters are not of the desired types (e.g. cmd_with_args should be a list)
+        OSError: Raises if timed_run is attempted to be used with gdb
+
+    Returns:
+        class: A rundata instance containing run information
+    """
+    if not isinstance(cmd_with_args, list):
+        raise TypeError("cmd_with_args should be a list (of strings).")
     if not isinstance(timeout, int):
         raise TypeError("timeout should be an int.")
 
-    useLogFiles = isinstance(logPrefix, str)  # pylint: disable=invalid-name
+    use_logfiles = isinstance(log_prefix, ("".__class__, u"".__class__))
```

I think this is wrong (has been since 025fa958abf97573c242f623d5144aec7ea2684e). We shouldn't silently ignore `log_prefix` if it is not a string, this should be an assertion and the checks for `use_logfiles` should be removed.

---

Removing the checks will happen in a follow-up PR. In the meantime, I will revert this line.